packages/x-data-grid/src/hooks/features/density/useGridDensity.tsx

```diff
@@ -43,15 +47,62 @@ export const useGridDensity = (
     }));
   });
 
-  React.useEffect(() => {
-    if (props.density) {
-      apiRef.current.setDensity(props.density);
-    }
-  }, [apiRef, props.density]);
-
   const densityApi: GridDensityApi = {
     setDensity,
   };
 
   useGridApiMethod(apiRef, densityApi, 'public');
+
+  /**
+   * PRE-PROCESSING
+   */
+  const stateExportPreProcessing = React.useCallback<GridPipeProcessor<'exportState'>>(
```

Comment:
We shouldn't be using `/**` comments for section annotations, those have a defined meaning and are picked-up by the LSP servers. The `stateExportPreProcessing` would have the comment above as its JSDoc comment.

Reply:
It feels wrong to use JSDoc comments that way. I think we should avoid it for new code and update the rest as we touch that code.